### ISSUE TITLE
[RFE]Request Tooltips be replaced with Info or ? icons

### ISSUE THREAD

**Issue by trumpy81** [WebTools-NG/WebTools-NG#553]
I would like to see the popup tooltips disappear, they are cumbersome and get in the way a lot of the time.

Most of the functions are clearly labelled and therefore do not benefit from a tooltip.

I would like to see either of these type icons to replace the tooltips if additional information is needed:
![1200px-Icon-round-Question_mark svg](https://user-images.githubusercontent.com/12381786/181644987-8f042cba-5ae8-4e82-bf15-d9f22987b626.png) ![2392583-200](https://user-images.githubusercontent.com/12381786/181645004-4c59b33f-a989-41ec-9750-888982ce12b4.png)

The information could popup on mouseover or only when the icon is clicked on.

Edit: Pages to update

* [x] ET
* [x] ET Settings
* [x] ET Custom
* [x] Plex.tv
* [x] Butler
* [x] FM
* [x] DVR
* [x] PMS Settings
* [x] View State
* [x] Global Settings
* [x] Global Settings/General
* [x] Global Settings/Export
* [x] Global Settings/PMS
* [x] Global Settings/Lib Mapping

**Comment by ukdtom** [comment]
@trumpy81 

On DB site, find the folder named 553

In there, is a homecooked build of Windows only

Install, and look at ET Settings page

You'll find 3 QMarks on the page

First two use hover, and the last one click action

So....

What to use?

**Comment by ukdtom** [comment]
Ohh....And other thoughts as well please ;)

**Comment by trumpy81** [comment]
I installed the build and looked at ET Settings. 

The icons are a little large and not centred in height, but otherwise OK. Maybe they should be more light grey than black though? 

The click on icon (bottom) is also click off, meaning the message stays visible until you click on it to turn it off. I'm not sure I like that, but it would suffice if need be.

The two mouse over versions are better since the popup only happens when you mouse over a small area, which means they don't get in the way and they are still convenient if you need to view/read that information.

So it looks like the mouse over icons get the nod from me ... lol

**Comment by ukdtom** [comment]
@trumpy81 

New builds avail on DB

There's one QMark that is misplaced on one page, and vertical center is also not fixed

For both are the reason, that I'm not skilled enough ;)

Maybe one day a pro dev will join us

**Comment by trumpy81** [comment]
There are a number of techniques you can use to adjust the vertical icon height.

Taking the General Settings page as an example, each text box is approx. 30 pixels in height. In that case, you need to increase the canvas size of the icon image to 30x30 pixels, but maintain the current image size within that canvas. It's easier done than said in Photoshop ... lol

You can place the image anywhere within that 30x30 pixel canvas.

A quick example of what I mean:
![Screenshot 2022-08-03 134221](https://user-images.githubusercontent.com/12381786/182520215-93cf063d-32f5-4cd3-a4ed-ab05e597a9ba.png) ![Screenshot 2022-08-03 134222](https://user-images.githubusercontent.com/12381786/182519926-884f24ba-c2d6-4e65-a60d-0eeab94a2165.png) ![Screenshot 2022-08-03 134223](https://user-images.githubusercontent.com/12381786/182519935-a509d00e-e7d1-4b66-be75-11aaf6dfcd5f.png)

I see the misplaced `?` icon on the Server Settings page. Perhaps you could place it to the right of the `Filter settings` title?

There seems to be an issue with the Select Type text boxes on the ExportTools and Find Media pages, they appear to be variable width instead of fixed width, see the image below:
![image](https://user-images.githubusercontent.com/12381786/182515260-96cc03d8-141f-4dbb-89ca-adb32a9a71c0.png)

**Comment by ukdtom** [comment]
To make life a tad more easy, I wrote a template that display the icon with the tooltip
That way, code only exists once, and is more easy to maintain

Sadly, on the Server Settings page, we need that for the label for a radio button group, and I can't figure out how to do that

And for now, it looks like I spent more time on the GUI than on actual code, so I need a break from the GUI ;)

**Comment by trumpy81** [comment]
Writing a class is a good idea, it does save on work down the track.

I am guessing that you would need to break into the radio button group class and insert your tool tip class in there somewhere after the title definition, but before the radio buttons definitions.

This is just an example:
```
    ShowMsgBox(message, type, title, ttid, buttons){
        const {remote} = require('electron'),
        dialog = remote.dialog,

        WIN = remote.getCurrentWindow();
        let options = {
            message: message,
            type: type,
            title: title,
            tooltip: ttid,
            buttons: buttons
        }
        return dialog.showMessageBoxSync(WIN, options);
    }
}
```
I have no idea if that would actually work or not, I am not familiar with electron etc., but hopefully, you get the idea?

Unfortunately, the GUI is important, because it is what everyone sees and interacts with, so no way to avoid it really and yes I do agree, it can get tiresome sometimes, but hang in there it will resolve itself sooner or later ... lol

**Comment by ukdtom** [comment]
> I am not familiar with electron

Heh, count me in as well, and add vue and node.js to that

**Comment by trumpy81** [comment]
I've been going through the code and I think I have identified the issue with the ? icon vertical height alignment.

`<WTNGtt tt="Modules.ET.Settings.ChNewLine_TT" size="20px"></WTNGtt>`

Try changing the above to `size=38px`

There may be places where you need to adjust that size to suit the elements they are attached to, but the size should not be too hard to figure out, I hope ... lol

I just hope it does not increase the icon size too much.

Another thought would be to add a margin top option to the wtng-tt.vue file.

```
    i.ttqmark{
        margin-top:5px:
        margin-left:5px;
    }
```
Adjust the pixel size as needed.

You can only try I guess :)

Try the following for the Server Settings (settings.vue) page:
```
<div> <!-- Settings to show -->
    <b-form-group id="FilterSettingsGroup" v-bind:label="$t('Modules.PMS.Settings.SettingsFilter')" label-size="lg" label-class="font-weight-bold pt-0">
    <WTNGtt tt="Modules.PMS.Settings.TTSettingsFilter" size="20px"></WTNGtt>
    <b-form-radio-group
        id="FilterSettings"
        v-model="selFilterSetting"
        @change.native="changeFilterSetting()"
        :options="FilterSettingsOptions"
        name="FilterSettings"
      >
      </b-form-radio-group>
    </b-form-group>
  </div>
```
With any luck, that should place the tooltip to the right of the title or underneath it, but above the radio buttons, fingers crossed, or it will screw things up completely ¯\_(ツ)_/¯ ... lol

**Comment by trumpy81** [comment]
I was worried that something like that would happen when using margin-top. 

Have you tried using the size I recommended (38px)?

BTW, I meant using different sizes on different pages, not different sizes on the same page.

I couldn't find a whole lot on v-bind, it seems they want you to use v-model, but that may only apply to web pages rendered in a browser?

Why can't we use BASIC, like we used to ... lol
